File: app/src/main/java/com/confessionsearchapptest/release1/data/notes/DeleteAsync.java
package com.confessionsearchapptest.release1.data.notes;

import android.os.AsyncTask;


public class DeleteAsync extends AsyncTask<Notes, Void, Void> {

    private final NoteDao mNoteDao;

    public DeleteAsync(NoteDao dao) {
        mNoteDao = dao;
    }

    @Override
    protected Void doInBackground(Notes... notes) {
        mNoteDao.delete(notes);
        return null;
    }
}

File: app/src/main/java/com/confessionsearchapptest/release1/data/notes/DeleteAsync.kt
package com.confessionsearchapptest.release1.data.notes

import android.os.AsyncTask
import com.confessionsearchapptest.release1.async.CoroutinesAsyncTask

class DeleteAsync(private val mNoteDao: NoteDao) : CoroutinesAsyncTask<Notes?, Void?, Void?>("delete") {
     override fun doInBackground(vararg notes: Notes?): Void? {
        mNoteDao.delete(*notes)
        return null
    }
}